src/Microsoft.ML.Data/Scorers/SchemaBindablePredictorWrapper.cs

```diff
@@ -497,14 +499,16 @@ public CalibratedRowMapper(RoleMappedSchema schema, SchemaBindableBinaryPredicto
                 }
             }
 
-            public Func<int, bool> GetDependencies(Func<int, bool> predicate)
+            /// <summary>
+            /// Given a set of columns, return the input columns that are needed to generate those output columns.
+            /// </summary>
+            IEnumerable<Schema.Column> IRowToRowMapper.GetDependencies(IEnumerable<Schema.Column> dependingColumns)
             {
-                for (int i = 0; i < OutputSchema.Count; i++)
-                {
-                    if (predicate(i) && InputRoleMappedSchema.Feature?.Index is int idx)
-                        return col => col == idx;
-                }
-                return col => false;
+
+                if (dependingColumns.Count() == 0 || !InputRoleMappedSchema.Feature.HasValue)
+                    return Enumerable.Empty<Schema.Column>();
+
+                return InputSchema.Where(col => col.Index == InputRoleMappedSchema.Feature.Value.Index);
```

Comment:
>InputSchema.Where(col => col.Index == InputRoleMappedSchema.Feature.Value.Index); [](start = 23, length = 81)

Here too.

Reply:
Duh! 
thanks :) 

---
In reply to: [256620362](https://github.com/dotnet/machinelearning/pull/2504#discussion_r256620362) [](ancestors = 256620362)